### ISSUE TITLE
Internal error when multiple tasks raise an exception; Trio backend only.

### ISSUE THREAD

**Issue by cyphase** [agronholm/anyio#474]
This only happens when using `anyio` with `trio`, whether via `anyio.run()` or `trio.run()`. It doesn't happen with the `asyncio` backend, or when using trio directly.

Reproducer:

```python
# CPython 3.10

import sys

import anyio  # 3.6.1
import trio  # 0.22.0 only. 0.21.0 and 0.20.0 work as expected.


async def task(task_id):
    raise Exception


async def anyio_main(num_tasks) -> None:
    async with anyio.create_task_group() as tg:
        for task_id in range(num_tasks):
            tg.start_soon(task, task_id)


async def trio_main(num_tasks) -> None:
    async with trio.open_nursery() as n:
        for task_id in range(num_tasks):
            n.start_soon(task, task_id)


if __name__ == "__main__":
    try:
        case = sys.argv[1].lower()
    except IndexError:
        case = "anyio_trio"

    try:
        num_tasks = int(sys.argv[2])
    except IndexError:
        num_tasks = 2

    if case == "anyio_asyncio":
        anyio.run(anyio_main, num_tasks, backend="asyncio")
    elif case == "anyio_trio":
        anyio.run(anyio_main, num_tasks, backend="trio")
    elif case == "trio":
        trio.run(trio_main, num_tasks)
```

```./repro.py anyio_trio``` gives this traceback:

```python
venv/lib/python3.10/site-packages/anyio/_backends/_trio.py:164: TrioDeprecationWarning: trio.MultiError is deprecated since Trio 0.22.0; use BaseExceptionGroup (on Python 3.11 and later) or exceptiongroup.BaseExceptionGroup (earlier versions) instead (https://github.com/python-trio/trio/issues/2211)
  class ExceptionGroup(BaseExceptionGroup, trio.MultiError):
venv/lib/python3.10/site-packages/anyio/_backends/_trio.py:188: TrioDeprecationWarning: trio.MultiError is deprecated since Trio 0.22.0; use BaseExceptionGroup (on Python 3.11 and later) or exceptiongroup.BaseExceptionGroup (earlier versions) instead (https://github.com/python-trio/trio/issues/2211)
  except trio.MultiError as exc:
Error in sys.excepthook:
Traceback (most recent call last):
  File "venv/lib/python3.10/site-packages/exceptiongroup/_formatting.py", line 71, in exceptiongroup_excepthook
    sys.stderr.write("".join(traceback.format_exception(etype, value, tb)))
  File "python3.10/traceback.py", line 135, in format_exception
    te = TracebackException(type(value), value, tb, limit=limit, compact=True)
  File "venv/lib/python3.10/site-packages/exceptiongroup/_formatting.py", line 114, in __init__
    for exc in exc_value.exceptions:
  File "venv/lib/python3.10/site-packages/exceptiongroup/_exceptions.py", line 89, in exceptions
    return tuple(self._exceptions)
AttributeError: 'NonBaseMultiError' object has no attribute '_exceptions'

Original exception was:
Traceback (most recent call last):
  File "repro.py", line 37, in <module>
    anyio.run(anyio_main, num_tasks, backend="trio")
  File "venv/lib/python3.10/site-packages/anyio/_core/_eventloop.py", line 70, in run
    return asynclib.run(func, *args, **backend_options)
  File "venv/lib/python3.10/site-packages/trio/_core/_run.py", line 2010, in run
    raise runner.main_task_outcome.error
  File "repro.py", line 12, in anyio_main
    async with anyio.create_task_group() as tg:
  File "venv/lib/python3.10/site-packages/anyio/_backends/_trio.py", line 189, in __aexit__
    raise ExceptionGroup(exc.exceptions) from None
trio.NonBaseMultiError: <exception str() failed>
```

**Comment by agronholm** [comment]
You should probably report this on the Trio issue tracker. This is not an AnyIO bug.

**Comment by agronholm** [comment]
I spoke too soon. I read the description more carefully now and indeed it did not happen with just trio.

**Comment by agronholm** [comment]
I've released v3.6.2 to pin the `trio` extra to `< 0.22`.
